codec/src/main/java/io/netty/handler/codec/json/JsonObjectDecoder.java

```diff
@@ -88,6 +90,11 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
             return;
         }
 
+        if (this.idx > in.readerIndex() && lastReaderIndex != in.readerIndex()) {
+            this.idx = in.readerIndex();
+            reset();
```

Comment:
It seemed to happen under heavy load, where a partial message was decoded then the call to discard made and by the next read the writer index was past it's previous position. The JsonDecoder then keeps reading looking for the correct closing bracket. Sometimes it finds one, by chance and returns a block with more than one JSON element in it, most times it gets stuck and never returns anything again.

I managed to replicate it reliably manually, but 'll try and replicate it in an automated test and update the PR.